### DISCUSSION TITLE
OTA Library for ESP-C3-12F and STM32L432

### DISCUSSION THREAD

**Discussion by Kamoool** [stm32duino/Arduino_Core_STM32#1777] [Libraries] [answered]
Hey,
I'm trying to find any library for OTA updating STM32L432 with ESP-C3-12F. Do you know of any?

**Comment by JAndrassy** [comment]
if you can add STM32L432 to my ArduinoOTA library. Only STM32F are supported now. It looks like L4 has flash organized as F3.

**Reply by Kamoool** [reply]
Ok, I'll try to do that. As I understand that's a library I should include in my ESP sketch? I would like it to function most of the time as a Serial <-> BLE interface, just when the special command is sent to turn off BT and enable Wifi for either ESP update (with BasicOTA sketch, but that's based on a different library)  or STM32 with a new part of the sketch. Your library writes to STM32 through Serial, right? I would just have to implement switching boot mode by BOOT0 and NRST pin, or wait for functionality in STM32 Core to jump to bootloader from sketch. Did I miss something?

**Reply by JAndrassy** [reply]
no. the library does a self update. for networking with esp32, you could use AT firmware 2.0.4 in esp32 and my WiFiEspAT library in STM32. Or if you want SPI between STM32 and esp32, then the nina-fw with WiFiNINA library is a good option.

**Reply by Kamoool** [reply]
I see, then your library should be implemented in STM32 sketch and use ESP32 just as a networking device to communicate over wifi? On ESP32 I should have [that](https://www.espressif.com/en/support/download/at)?

Do you see a way of doing what I mentioned, but with your library? STM32 + your ArduinoOTA, connected to ESP32-C3 (by serial) running BasicOTA with disabled WiFi most of the time but enabled BT. Then the specific command is sent through BT, it stops BT, starts Wifi, and starts polling for either STM32 update or self-update with BasicOTA part.

**Comment by JAndrassy** [comment, marked as answer]
> On ESP32 I should have [that](https://www.espressif.com/en/support/download/at)?

yes

BasicOTA for esp32 updates the esp32.
arduinoOTA tool from ArduinoIDE  does a http upload of the binary. but then the binary needs to be send to stm32 bootloader.  what tool does a normal serial upload to STM32L4? that tool mut be implemented on esp32 then.
